mem/addr_book.go

```diff
@@ -127,31 +95,28 @@ func (mgr *AddrManager) AddAddrs(p peer.ID, addrs []ma.Multiaddr, ttl time.Durat
 		if !found || exp.After(a.Expires) {
 			amap[addrstr] = expiringAddr{Addr: addr, Expires: exp, TTL: ttl}
 
-			mgr.subManager.BroadcastAddr(p, addr)
+			mab.subManager.BroadcastAddr(p, addr)
 		}
 	}
 	newAddrs := make([]expiringAddr, 0, len(amap))
 	for _, ea := range amap {
 		newAddrs = append(newAddrs, ea)
 	}
-	mgr.addrs[p] = newAddrs
+	mab.addrs[p] = newAddrs
 }
 
 // SetAddr calls mgr.SetAddrs(p, addr, ttl)
-func (mgr *AddrManager) SetAddr(p peer.ID, addr ma.Multiaddr, ttl time.Duration) {
-	mgr.SetAddrs(p, []ma.Multiaddr{addr}, ttl)
+func (mab *memoryAddrBook) SetAddr(p peer.ID, addr ma.Multiaddr, ttl time.Duration) {
+	mab.SetAddrs(p, []ma.Multiaddr{addr}, ttl)
 }
 
 // SetAddrs sets the ttl on addresses. This clears any TTL there previously.
 // This is used when we receive the best estimate of the validity of an address.
-func (mgr *AddrManager) SetAddrs(p peer.ID, addrs []ma.Multiaddr, ttl time.Duration) {
-	mgr.addrmu.Lock()
-	defer mgr.addrmu.Unlock()
-
-	// so zero value can be used
-	mgr.init()
```

Comment:
init

Reply:
No longer needed as the `AddrManager/AddrBook` is a private struct with a public constructor `NewAddrBook()`. The init is only done once now, as opposed to checked with every call like before.